packages/onchainkit/src/fund/components/FundCardSubmitButton.tsx

```diff
@@ -34,17 +40,79 @@ export function FundCardSubmitButton() {
     [fundAmountCrypto, fundAmountFiat],
   );
 
+  const buttonIcon = useMemo(() => {
+    switch (submitButtonState) {
+      case 'loading':
+        return '';
+      case 'success':
+        return <SuccessSvg fill="#F9FAFB" />;
+      case 'error':
+        return <ErrorSvg fill="#F9FAFB" />;
+      default:
+        return null;
+    }
+  }, [submitButtonState]);
+
+  const buttonTextContent = useMemo(() => {
+    switch (submitButtonState) {
+      case 'loading':
+        return '';
+      case 'success':
+        return 'Success';
+      case 'error':
+        return 'Something went wrong';
+      default:
+        return 'Buy';
+    }
+  }, [submitButtonState]);
+
+  const buttonContent = useMemo(() => {
+    if (submitButtonState === 'loading') {
+      return <Spinner />;
+    }
+
+    return (
+      <>
+        {buttonIcon && (
+          <span
+            data-testid="ockFundButtonIcon"
+            className="flex h-6 items-center"
+          >
+            {buttonIcon}
+          </span>
+        )}
+        <span data-testid="ockFundButtonTextContent">{buttonTextContent}</span>
+      </>
+    );
+  }, [submitButtonState, buttonIcon, buttonTextContent]);
+
+  // FundButton accepts render prop or children but not both
+  if (render) {
+    return (
+      <FundButton
+        disabled={isButtonDisabled}
+        className="w-full"
```

Comment:
nit: Since this is passing render down to FundButton, this className won't get rendered